src/smart-components/group/principal/add-group-members.js

```diff
@@ -1,57 +1,55 @@
 import React, { useState } from 'react';
-import { withRouter } from 'react-router-dom';
-import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
+import { useDispatch } from 'react-redux';
+import { useHistory, useRouteMatch } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import { Button, Modal, ModalVariant, StackItem, Stack, TextContent } from '@patternfly/react-core';
 import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/';
-import { addGroup, addMembersToGroup, fetchMembersForGroup, fetchGroups } from '../../../redux/actions/group-actions';
+import { addMembersToGroup, fetchMembersForGroup, fetchGroups } from '../../../redux/actions/group-actions';
 import UsersList from '../add-group/users-list';
 import ActiveUser from '../../../presentational-components/shared/ActiveUsers';
 import { useIntl } from 'react-intl';
 import messages from '../../../Messages';
 
-const AddGroupMembers = ({
-  history: { push },
-  match: {
-    params: { uuid },
-  },
-  addNotification,
-  closeUrl,
-  addMembersToGroup,
-  fetchMembersForGroup,
-  fetchGroups,
-}) => {
+const AddGroupMembers = ({ closeUrl }) => {
   const [selectedUsers, setSelectedUsers] = useState([]);
   const intl = useIntl();
+  const { push } = useHistory();
+  const uuid = useRouteMatch('/groups/detail/:uuid') ? useRouteMatch('/groups/detail/:uuid').params.uuid : null;
```

Comment:
@Hyperkid123 Not 100% sure on this one, but I was having reference issues when I left `match` above. This fixed those reference issues as well as the tests that were failing. I reference how `useRouteMatch()` was being used in `edit-group-modal.js`

Reply:
You can get the `match` from `useRouteMatch`. But you can't have hooks in conditions. You want to get the match object and then attempt to find the `uuid`.
```diff
- const uuid = useRouteMatch('/groups/detail/:uuid') ? useRouteMatch('/groups/detail/:uuid').params.uuid : null;
+ const match = useRouteMatch('/groups/detail/:uuid')
+ const uuid = match?.params.uuid
```
But I would consider using `useParams` hook and look foe the `uuid` param there. That hook should give you all params from URl. But if the route mask is not correct it might not work.